src/System.Private.CoreLib/shared/System/MemoryExtensions.cs

```diff
@@ -1653,14 +1654,17 @@ public static bool Overlaps<T>(this ReadOnlySpan<T> span, ReadOnlySpan<T> other,
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static int BinarySearch<T, TComparer>(
             this ReadOnlySpan<T> span, T value, TComparer comparer)
-            where TComparer : IComparer<T>
+            where TComparer : IComparer<T>?
         {
+
             if (comparer == null)
-                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.comparer);
+            {
+                return BinarySearch(span,
+                    new SpanHelpers.ComparerComparable<T, Comparer<T>>(value, Comparer<T>.Default));
```

Comment:
nit: Can we store the result in a local and avoid having two return statements and method calls (and only do a single call)?

Reply:
I don't think we can do that because each instance of `ComparerComparable` are of different type, so I couldn't do it without hurting the generic specialization for structs. One instance is of type `ComparerComparable<T, Comparer<T>>` and the other one is of type `ComparerComparable<T, TComparer>`.

Reply:
This means that we will create a unnecessary generic instantiation. It is pile of bloat that is not used. I do not think this is a good change.

Reply:
So you think we shouldn't support a nullable comparer here?

Reply:
Yes, I do not think we should support nullable comparer here. It just adds footprint, without adding value.

Reply:
Ok, in API Review we were trying to be consistent with APIs that take a comparer. I will then revert that change if it is not worth taking the bloat.

Reply:
Reverted it.